check_runner/__init__.py

```diff
@@ -0,0 +1,153 @@
+    default=[]
+)
+parser.add_argument(
+    '--debug',
+    help='display debug messages',
+    default=False,
+    action='store_true'
+)
+
+codes_mapping = {
+    nagios_common.Codes.OK: 1,
+    nagios_common.Codes.WARNING: 4,
+    nagios_common.Codes.CRITICAL: 4,
+    nagios_common.Codes.UNKNOWN: None
+}
+
+
+def main(script, component_id, config_file, script_args=None, retries=5, interval=0.5, debug=False):
+    """Script execution"""
+
+    if script_args is None:
+        script_args = []
+
+    logging.basicConfig(level=logging.INFO)
+    logger = logging.getLogger(__name__)
+
+    if debug: # pragma: no cover
+        logger.setLevel(logging.DEBUG)
+        logger.info('debug logging enabled')
+
+    # Read Cachet API parameters from config file
+    config_parser = settings.RunnerConfigParser()
+    config_parser.read(config_file)
+    base_url = config_parser.get('Cachet', 'base-url', fallback=None)
+    api_key = config_parser.get('Cachet', 'api-key', fallback=None)
+    logger.info('Config params: base-url = %s, api-key = %s', base_url, api_key)
+    if not api_key or not base_url:
+        logger.critical('one or both mandatory config parameters are missing')
+        raise ValueError('Invalid config file, use config.ini.dist file as reference')
+
+    # Run monitoring script
+    script_args.insert(0, script)
+    logger.debug('invocation arguments %s', script_args)
+    script_return_code = None
+    remaining_attempts = retries
+    while script_return_code != nagios_common.Codes.OK and remaining_attempts >= 0:
+        if remaining_attempts < retries:
+            # Not first run, script is in soft failure status, wait before retrying
+            time.sleep(interval)
+        logger.info('attempt n. %d', retries - remaining_attempts + 1)
+
+        completed_execution = subprocess.run(script_args)
+        logger.debug('execution completed, plugin exit code is %d', completed_execution.returncode)
+        logger.debug('type of plugin exit code is %s', completed_execution.returncode)
+        try:
+            script_return_code = nagios_common.Codes(completed_execution.returncode)
+        except ValueError:
+            logger.critical('script plugin exit code is not compatible with Nagios standards, '
+                            'return code is %d', completed_execution.returncode)
+            nagios_common.plugin_exit(code=nagios_common.Codes.CRITICAL)
+
+        logger.info('script return code is: %s, %d', script_return_code.name,
+                    script_return_code.value)
+        remaining_attempts -= 1
+
+    # Update Cachet
+    cachet_status_code = codes_mapping[script_return_code]
+    if cachet_status_code:
+        components = cachet.Components(endpoint=base_url, api_token=api_key)
+        components.put(id=component_id, status=cachet_status_code)
+
+        logger.info('updated component %d with status %d', component_id, cachet_status_code)
+        nagios_common.plugin_exit(code=nagios_common.Codes.OK)
```

Comment:
Seems weird to use nagios *plugin* logic to make the *runner* exit.

Reply:
It does make more sense since it makes a difference if the `runner` exits with different codes, hence instead of using `exit(50000)` I felt it made more sense to have it strictly and explicitly abide to the standard the whole project is based on.

Reply:
Fair enough.

Reply:
Just for the records: `check_runner` is a Nagios plugin itself and can be used as such.